tst/com/amazon/corretto/crypto/provider/test/AesKeyWrapTest.java

```diff
@@ -0,0 +1,71 @@
+package com.amazon.corretto.crypto.provider.test;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+import static org.junit.jupiter.api.Assumptions.assumeTrue;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.security.GeneralSecurityException;
+import java.security.InvalidAlgorithmParameterException;
+import java.security.InvalidKeyException;
+import java.security.Key;
+import java.security.SecureRandom;
+import java.util.Arrays;
+import java.util.Iterator;
+import java.util.Spliterator;
+import java.util.Spliterators;
+import java.util.stream.Stream;
+import java.util.stream.StreamSupport;
+import java.util.zip.GZIPInputStream;
+
+import javax.crypto.AEADBadTagException;
+import javax.crypto.BadPaddingException;
+import javax.crypto.Cipher;
+import javax.crypto.IllegalBlockSizeException;
+import javax.crypto.SecretKey;
+import javax.crypto.spec.GCMParameterSpec;
+import javax.crypto.spec.SecretKeySpec;
+
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.api.parallel.Execution;
+import org.junit.jupiter.api.parallel.ExecutionMode;
+import org.junit.jupiter.api.parallel.ResourceAccessMode;
+import org.junit.jupiter.api.parallel.ResourceLock;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.MethodSource;
+
+@ExtendWith(TestResultLogger.class)
+@Execution(ExecutionMode.CONCURRENT)
+@ResourceLock(value = TestUtil.RESOURCE_GLOBAL, mode = ResourceAccessMode.READ)
+public final class AesKeyWrapTest {
```

Comment:
Your next critical tests are:
* Keys of lengths: {3, 4, 7, 8, 9, 15, 16, 17, 1023, 1024, 1025}
* Compatibility with results from the stock Java implementation (or BC)

Reply:
Those compatibility tests really need to come now. They are the most important ones you'll add.

Reply:
got it. working on 'em.

> Keys of lengths: {3, 4, 7, 8, 9, 15, 16, 17, 1023, 1024, 1025}

should we just be testing these as AES keys? or also as RSA, EC, etc.?

Reply:
They aren't valid keys of any lengths. So, use type "Generic" and have them be `SecretKey`s.